src/test/scala/io/iohk/ethereum/mining/BlockGeneratorSpec.scala

```diff
@@ -92,6 +92,27 @@ class BlockGeneratorSpec extends FlatSpec with Matchers with PropertyChecks with
     fullBlock.right.foreach(b => ledger.executeBlock(b, blockchainStorages.storages, validators) shouldBe a[Right[_, Seq[Receipt]]])
   }
 
+  it should "include transaction with higher gas price if nonce is the same"in new TestSetup {
+    val txWitSameNonceButLowerGasPrice: SignedTransaction = SignedTransaction.sign(
+      transaction.copy(gasPrice = signedTransaction.tx.gasPrice - 1),
+      keyPair,
+      Some(0x3d.toByte))
+
+    val result: Either[BlockPreparationError, PendingBlock] =
+      blockGenerator.generateBlockForMining(1, Seq(txWitSameNonceButLowerGasPrice, signedTransaction), Nil, Address(testAddress))
+    result shouldBe a[Right[_, Block]]
+
+    //mined with etc-client + ethminer
+    val minedNonce = ByteString(Hex.decode("5e8d5c12cea7e0c7"))
+    val minedMixHash = ByteString(Hex.decode("9247b81258f97159f987a5f4f9e94df1d95e10eeabff2836020eafb27a8228b0"))
+    val miningTimestamp = 1494604913
+
+    val fullBlock: Either[BlockPreparationError, Block] = result.right
```

Comment:
Minor detail: Should we maybe include a check to see that `signedTransaction` is included in the `fullBlock` and not `txWitSameNonceButLowerGasPrice`?